File: backend/utils/dates.py
"""Indian Financial Year utilities (April 1 - March 31)."""
from datetime import date, datetime


def get_indian_financial_year(reference_date: date = None) -> dict:
    if reference_date is None:
        reference_date = date.today()
    year = reference_date.year
    month = reference_date.month
    fy_start_year = year if month >= 4 else year - 1
    fy_end_year = fy_start_year + 1
    return {
        "start": datetime(fy_start_year, 4, 1),
        "end": datetime(fy_end_year, 3, 31, 23, 59, 59),
        "label": f"FY {fy_start_year}-{str(fy_end_year)[2:]}",
        "start_year": fy_start_year,
        "end_year": fy_end_year,
    }

File: frontend/src/utils/dates.js
/**
 * Indian Financial Year utilities (April 1 - March 31)
 */

export const getIndianFinancialYear = (date = new Date()) => {
  const year = date.getFullYear();
  const month = date.getMonth() + 1;
  const fyStartYear = month >= 4 ? year : year - 1;
  const fyEndYear = fyStartYear + 1;
  return {
    start: new Date(fyStartYear, 3, 1),
    end: new Date(fyEndYear, 2, 31),
    label: `FY ${fyStartYear}-${String(fyEndYear).slice(2)}`,
  };
};

export const getCurrentFYStart = () => getIndianFinancialYear().start;
export const getCurrentFYEnd = () => getIndianFinancialYear().end;
export const getCurrentFYLabel = () => getIndianFinancialYear().label;
